python/cudf/cudf/core/dataframe.py

```diff
@@ -610,6 +610,192 @@ class _DataFrameiAtIndexer(_DataFrameIlocIndexer):
     pass
 
 
+@_performance_tracking
+def _listlike_to_column_accessor(
+    data: Sequence,
+    columns: None | pd.Index,
+    index: None | cudf.Index,
+    nan_as_null: bool,
+) -> tuple[dict[Any, ColumnBase], cudf.Index, pd.Index]:
+    """
+    Convert a list-like to a dict for ColumnAccessor for DataFrame.__init__
+
+    Returns
+    -------
+    tuple[dict[Any, ColumnBase], cudf.Index, pd.Index]
```

Comment:
Maybe worth noting that the last return value is a pandas index because we always use a host index for the columns because of the nature of the iteration patterns and the associated performance (and point to anywhere else in the code that we document this behavior).

Reply:
Sure thing. Added a little note